packages/editor/src/components/post-featured-image/get-featured-media-details.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * WordPress dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+
+/**
+ * The media details object.
+ *
+ * @typedef {Object} MediaDetails
+ *
+ * @property {string} mediaWidth     Width value of the media.
+ * @property {string} mediaHeight    Height value of the media.
+ * @property {string} mediaSourceUrl URL of the media.
+ */
+
+/**
+ * Returns the featured media dimensions and source URL.
+ *
+ * @param {Object} media  The media object.
+ * @param {number} postId The post ID of the media
+ * @return {MediaDetails} The featured media details.
+ */
+export default function getFeaturedMediaDetails( media, postId ) {
```

Comment:
It is very unclear to me why this is a selector of "editor" package instead of being a "hook" in edit-site for instance. `editor` should never be imported in edit-site IMO.

Reply:
Looking further at the PR, I guess we want to reuse the same functionality that was present in this component. I think "core-data" is a more suited place for this and I'm also not sure whether it should be a hook or selector.

Reply:
There is the option to revert this entirely and use the medium image size, and where that's not available the full one in the sidebar. 

The image dimensions are constrained and cropped by CSS anyway so, image download size aside, the width and height are not as important since we're not displaying the entire image as we are in the block inspector.